web/src/app/WebApp/Installers/Flarum/FlarumSetup.php

```diff
@@ -0,0 +1,211 @@
+<?php
+namespace Hestia\WebApp\Installers\Flarum;
+
+use Hestia\System\Util;
+use Hestia\WebApp\Installers\BaseSetup as BaseSetup;
+
+class FlarumSetup extends BaseSetup {
+	protected $appInfo = [
+		"name" => "Flarum",
+		"group" => "forum",
+		"enabled" => true,
+		"version" => "latest",
+		"thumbnail" => "fl-thumb.png",
+	];
+
+	protected $appname = "flarum";
+
+	protected $config = [
+		"form" => [
+			"forum_title" => ["type" => "text", "value" => "Flarum Forum"],
+			"admin_username" => ["value" => "fladmin"],
+			"admin_email" => "text",
+			"admin_password" => "password",
+			"install_directory" => ["type" => "text", "value" => "", "placeholder" => "/"],
+		],
+		"database" => true,
+		"resources" => [
+			"composer" => ["src" => "flarum/flarum"],
+		],
+		"server" => [
+			"php" => [
+				"supported" => ["8.0", "8.1", "8.2"],
+			],
+		],
+	];
+
+	// Our updateFile routine done the 'Hestia way'
+	public function updateFile($file, $search, $replace) {
+		$result = null;
+		$this->appcontext->runUser("v-open-fs-file", [$file], $result);
+		foreach ($result->raw as $line_num => $line) {
+			if (strpos($line, $search) !== false) {
+				$result->raw[$line_num] = str_replace($search, $replace, $line);
+			}
+		}
+		$tmp = $this->saveTempFile(implode("\r\n", $result->raw));
+		if (!$this->appcontext->runUser("v-move-fs-file", [$tmp, $file], $result)) {
+			throw new \Exception("Error updating file in: " . $tmp . " " . $result->text);
+		}
+		return $result;
+	}
+
+	public function install(array $options = null): bool {
+		parent::setAppDirInstall($options["install_directory"]);
+		parent::install($options);
+		parent::setup($options);
+		$result = null;
+
+		// Move public folder content (https://docs.flarum.org/install/#customizing-paths)
```

Comment:
From a Flarum POV, this type of install should be avoided if at all possible.

As mentioned in a comment, we have an nginx config available which can handle path rewriting automatically so this shouldn't be needed.